trin-history/src/network.rs

```diff
@@ -65,53 +64,17 @@ impl HistoryNetwork {
         // The overlay ping via talkreq will trigger a session at the base layer, then
         // a session on the (overlay) portal network.
         for enr in self.overlay.discovery.discv5.table_entries_enr() {
-            debug!("Pinging {} on portal history network", enr);
+            debug!("Attempting bond with bootnode {}", enr);
             let ping_result = self.overlay.send_ping(enr.clone()).await;
 
             match ping_result {
                 Ok(_) => {
                     debug!("Successfully bonded with {}", enr);
                     continue;
                 }
-                Err(OverlayRequestError::ChannelFailure(error)) => {
-                    debug!("Channel failure sending ping: {}", error);
-                    continue;
-                }
-                Err(OverlayRequestError::Timeout) => {
-                    debug!("Timed out while bonding with {}", enr);
-                    continue;
-                }
-                Err(OverlayRequestError::EmptyResponse) => {
-                    debug!("Empty response to ping from: {}", enr);
-                    continue;
-                }
-                Err(OverlayRequestError::InvalidRequest(_)) => {
-                    debug!("Sent invalid ping request to {}", enr);
-                    continue;
-                }
-                Err(OverlayRequestError::InvalidResponse) => {
-                    debug!("Invalid ping response from: {}", enr);
-                    continue;
-                }
-                Err(OverlayRequestError::Failure(_)) => {
-                    debug!("Failure to serve ping response from: {}", enr);
-                    continue;
-                }
-                Err(OverlayRequestError::DecodeError) => {
-                    debug!("Error decoding ping response from: {}", enr);
-                    continue;
-                }
-                Err(OverlayRequestError::AcceptError(error)) => {
-                    debug!("Error building Accept message: {:?}", error);
-                }
-                Err(OverlayRequestError::Discv5Error(error)) => {
-                    debug!("Unexpected error while bonding with {} => {:?}", enr, error);
-                    return Err(anyhow!(error.to_string()));
-                }
-                _ => {
-                    let msg = format!("Unexpected error while bonding with {enr}");
-                    debug!("{msg}");
-                    return Err(anyhow!(msg));
+                Err(err) => {
+                    error!("{err} while pinging bootnode: {enr:?}");
+                    std::process::exit(1);
```

Comment:
My bad, I should've been a bit more descriptive with my words. By "panicking" I didn't necessarily mean an explicit panic, but just return an error which will cause a panic when in gets unwrapped [here](https://github.com/ethereum/trin/blob/master/trin-history/src/lib.rs#L100)

Reply:
Yea but regardless of where we panic (explicitly or due to unwrap or error) it will still appear to the user as an error on our end, rather than an issue with the bootnode. With this approach we get a clean end of the program. There's a crate called `human-panic` that could end our program nicely but that seems like overkill compared to this approach.